openapi.yaml

```diff
@@ -469,1250 +491,1393 @@ paths:
+        - `RK` - Financial Aid Address
+        - `VO` - Vendor Order
+        - `VP` - Vendor Payment
+        - `VF` - Vendor Fax
+        - `CM` - Current
+        - `EO` - Employee Office (Directory)
+        - `RH` - Residence Hall
+        - `EF` - Employee Office Fax
+        - `MP` - Mobile Phone
+      schema:
         type: string
-        format: date
-        description: The date that the employee accepted or denied to have their salary information shared.
-        example: '1970-01-01'
-      retirement:
-        type: object
-        properties:
-          code:
-            type: string
-            description: Retirement code
-      i9Form:
-        type: object
-        properties:
-          code:
-            type: string
-            description: |
-              Code to indicate the status of the employee's I9 form.
-              - `R` - Received
-              - `N` - Not Received
-              - `T` - Temporary
-              - `E` - Exempt
-      laborDistribution:
-        type: array
-        items:
+        enum:
+          &phoneTypeEnums [PA, RN, RB, MG, RE, RK, VO, VP, VF, CM, EO, RH, EF, MP]
+  responses:
+    '400':
+      description: Bad request
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorResult'
+    '404':
+      description: Resource not found
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorResult'
+    '409':
+      description: Type or ID conflict
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorResult'
+    '500':
+      description: Internal server error
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorResult'
+  securitySchemes:
+    OAuth2:
+      type: oauth2
+      flows:
+        clientCredentials:
+          tokenUrl: https://api.oregonstate.edu/oauth2/token
+          scopes:
+            full: Full access to the API
+  schemas:
+    PersonType:
+      type: string
+      enum: [person]
+    JobType:
+      type: string
+      enum: [job]
+    SsnType:
+      type: string
+      enum: [ssn]
+    MealPlanType:
+      type: string
+      enum: [meal-plans]
+    AddressType:
+      type: string
+      enum: [addresses]
+    PhoneType:
+      type: string
+      enum: [phones]
+    EmailType:
+      type: string
+      enum: [emails]
+    PersonPostRequired:
+      type: object
+      required:
+        - firstName
+        - lastName
+        - birthDate
+        - sex
+    PersonPostAttributes:
+      type: object
+      additionalProperties: false
+      properties:
+        firstName:
+          type: string
+          description: First name of person
+          example: John
+        lastName:
+          type: string
+          description: Last name of person
+          example: Smith
+        middleName:
+          type: string
+          nullable: true
```

Comment:
Nice. Do you think we should also explicitly mark those required fields be `nullable: false`?

Reply:
Not sure. Doesn't look like its worth the effort to me.

Reply:
We can definitely have our integration test or validation logic only valid the data type only if `nullable` is set to `true`. I'm thinking about this from the UI wise. As the screenshot attached, the `nullable` attribute will only show if you specify it.

![Screen Shot 2020-03-11 at 10 01 34 AM](https://user-images.githubusercontent.com/9169491/76443235-59fd7280-637f-11ea-8d1a-393b9d3194de.png)

There's a red star specified required fields so it should be fine for request body, however, it can be a little confusing in response schema.

![Screen Shot 2020-03-11 at 10 06 32 AM](https://user-images.githubusercontent.com/9169491/76443643-02133b80-6380-11ea-846f-efac1ccb9544.png)

Reply:
I added nullable to all fields